### ISSUE TITLE
Guardrails Test: Write test to check policies associated to account- 1st draft

### ISSUE THREAD

**Issue by crochcunill** [bcgov/cloud-pathfinder#1702]
**Describe the issue**
The following AWS CLI command

> $ aws iam get-account-authorization-details

produces a JSON file that contains all the policies associated to an account.

A possible way to test the policies have not been altered by a SEA update/upgrade is to compare the results of the previous command before and after the update/upgrade

**Additional context**
The Service Control Policies added by the AWS SEA (https://github.com/aws-samples/aws-secure-environment-accelerator/tree/main/reference-artifacts/SCPs) are not included in the previous file. Need to determine if it makes sense to also compare them, and if they need to be monitored, the best way to do it, as currently they only appear explicitly in the core accounts and not in the workload accounts.

Also, need to monitor if after and upgrade/update something changes. For example, policies have dates associated to them and very minor changes may change the date, so a direct comparison between the before/after files may not be the most appropriate action. Will need to look for other ways, perhaps checking a % of lines are identical? perhaps checking node by node?...

- We probably will not go deep into the CLI path, however, this is still worth doing because we need to check our other more scalable and GUI-based test automation paths that we come up with for completeness and correctness.  This work will be a good baseline to make sure something like CheckPoint CPSM is working as we expect.

**Definition of done**
- A command line script that will run the get-account-authorization-details command, compare (whatever the chosen strategy) with the baseline file and decide if there is a problem
- If the produced file before/after are identical, then it will be a very simple script. If they are not, and there are minor differences, then it may became way more complex to separate irrelevant changes from the important ones.

**Comment by crochcunill** [comment]
See #1575 and #1765:  the current task (#1702) is build on top of these two tasks

**Comment by crochcunill** [comment]
While working on ticket #1765 I created scripts to extract information from the files returned by Cloudguard and AWS. These files are stored at https://github.com/crochcunill/parsing_JSON_GC_AWS_files (public repo)

The scripts have good bones but they need a fair amount of polishing. I have stopped working on them because some of the results from Cloudguard do not seem to appear in AWS and vice versa. I want to refine my understanding of what Cloudguard is returning before bing able to compare the results of Cloudguard with AWS

However, it is possible to use the current scripts to analyze and compare the files returned by AWS **or** Cloudguard before/after upgrade, with this test being the most basic test to check the integrity of the guardrails

Check this document https://bcgov.sharepoint.com/:b:/t/01368-CPFScrumTeam/EcODx2WpWmZOmKa4lnZiS8IBec8nuQm4rbYMVuOk7fnvww?e=Rn7FzA for the test cases

**Comment by crochcunill** [comment]
While working on ticket #1765 I created scripts to extract information from the files returned by Cloudguard and AWS. These files are stored at https://github.com/crochcunill/parsing_JSON_GC_AWS_files (public repo)

The scripts have good bones but they need a fair amount of polishing. I have stopped working on them because some of the results from Cloudguard do not seem to appear in AWS and vice versa. I want to refine my understanding of what Cloudguard is returning before bing able to compare the results of Cloudguard with AWS

However, it is possible to use the current scripts to analyze and compare the files returned by AWS **or** Cloudguard before/after upgrade, with this test being the most basic test to check the integrity of the guardrails

Check this document https://bcgov.sharepoint.com/:b:/t/01368-CPFScrumTeam/EcODx2WpWmZOmKa4lnZiS8IBec8nuQm4rbYMVuOk7fnvww?e=Rn7FzA for the test cases
